demos-europe/demosplan-core
# F t29878 disconnect addon data storage

**Ticket:** https://yaits.demos-deutschland.de/T29878

Description:
- Link CoreEntities to AddOn Interfaces on core
- Instead of linking every entity manually to its corresponding Interface in doctrine.yaml, the DemosPlanKernel does it using using AddonResolveTargetEntity (which is a CompilerPass)
- AddonResolveTargetEntity goes through the folder "Entity" in core and detects which Entities implement AddOn interfaces. If they do, then link them using addResolveTargetEntity.

### How to review/test
- Everytime that we create a new class in core (which implements an Addon Interfaces) we should clear cache e.g. bin/ewm dp:c:c , so that the DemosPlanKernel and consequently the new entities are linked with its interfaces.
- You can install demospipe:
-- in DemosEurope\DemosplanAddon\DemosPipes\Entity\PdfImport\AnnotatedStatementPdf update demosplan\DemosPlanCoreBundle\Entity\Procedure\Procedure for DemosEurope\DemosplanAddon\Contracts\Entities\ProcedureInterface
- Test DemosPipes

Delete the checkbox if it doesn't apply/isn't necessary.

- [x ] Move the tickets on the board accordingly
